python/tests/integration/toolbox/test_library_tool.py

```diff
@@ -230,3 +232,92 @@ def iterate_through_version_chain(key):
     assert len(keys_by_key_type[KeyType.TABLE_DATA]) == (num_versions-1) % 3 + 1
     assert len(keys_by_key_type[KeyType.TOMBSTONE_ALL]) == num_versions // 3
 
+
+def test_overwrite_append_data(object_and_mem_and_lmdb_version_store):
+    lib = object_and_mem_and_lmdb_version_store
+    if lib._lib_cfg.lib_desc.version.encoding_version == 1:
```

Comment:
Just checking this is definitely zero indexed and we aren't skipping on v1 encoding by mistake?

Reply:
Yes it's [zero indexed](https://github.com/man-group/ArcticDB/blob/b20a6776b02e9c8120d748e85d648c26b59f0a33/cpp/arcticdb/storage/memory_layout.hpp#L114-L116).

Can also be seen in the CI logs (just v2 is skipped)